pkgs/development/python-modules/cmsis-pack-manager/default.nix

```diff
@@ -1,78 +1,58 @@
 { lib
 , fetchPypi
 , rustPlatform
+, cffi
+, libiconv
 , stdenv
-, Security
-, writeShellScriptBin
+, darwin
 , buildPythonPackage
-, setuptools-scm
 , appdirs
-, milksnake
 , pyyaml
 , hypothesis
 , jinja2
-, mock
 , pytestCheckHook
+, unzip
 }:
-let
-  pname = "cmsis-pack-manager";
-  version = "0.4.0";
+
+buildPythonPackage rec {
+  pname = "cmsis_pack_manager";
+  version = "0.5.2";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-NeUG6PFI2eTwq5SNtAB6ZMA1M3z1JmMND29V9/O5sgw=";
+    hash = "sha256-sVfyz9D7/0anIp0bEPp1EJkERDbNJ3dCcydLbty1KsQ=";
   };
 
-  native = rustPlatform.buildRustPackage {
-    name = "${pname}-${version}-native";
-
+  cargoDeps = rustPlatform.fetchCargoTarball {
     inherit src;
-
-    buildInputs = lib.optionals stdenv.isDarwin [
-      Security
-    ];
-
-    sourceRoot = "${pname}-${version}/rust";
-    cargoLock.lockFile = ./Cargo.lock;
-
-    postPatch = ''
-      cp ${./Cargo.lock} Cargo.lock
-    '';
-
-    cargoBuildFlags = [ "--lib" ];
+    sha256 = "dO4qw5Jx0exwb4RuOhu6qvGxQZ+LayHtXDHZKADLTEI=";
   };
-in
-buildPythonPackage rec {
-  inherit pname version src;
-
-  # The cargo build is already run in a separate derivation
-  postPatch = ''
-    substituteInPlace setup.py \
-        --replace "'cargo', 'build'," "'true',"
-  '';
-
-  nativeBuildInputs = [ setuptools-scm ];
-  propagatedBuildInputs = [ appdirs milksnake pyyaml ];
 
-  nativeCheckInputs = [ hypothesis jinja2 mock pytestCheckHook ];
+  nativeBuildInputs = [ rustPlatform.cargoSetupHook rustPlatform.maturinBuildHook ];
+  propagatedNativeBuildInputs = [ cffi ];
```

Comment:
Is there any particular reason `cffi` is added to `propagatedNativeBuildInputs` instead of `propagatedBuildInputs`?

It's something that's needed on the host platform and right now `nix run nixpkgs#pyocd` complains about missing cffi because this causes it to not be included in the wrapper

I don't use `pyocd`, but someone asked about it in Matrix and I happened to notice this and also noticed that moving this to `propagatedBuildInputs` fixes `nix run nixpkgs#pyocd`